.github/workflows/release.yaml

```diff
@@ -1,21 +1,48 @@
+---
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
 name: goreleaser
+
 on:
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: "Dry Run"
+        default: false
+        required: false
+        type: boolean
+
   push:
-    tags:
-      - "*"
+    tags: ["*"]
+
+  pull_request:
+    paths:
+      - ".github/workflows/release.yaml"
+      - ".goreleaser.yaml"
+      - "go.mod"
+      - "go.sum"
+      - "cmd/**"
+
+env:
+  GORELEASER_DRY_RUN: "${{ inputs.dry-run || github.event_name == 'pull_request' }}"
```

Comment:
I don't know much about GH Action, but giving a boolean value to env var will set the variable to the string value of the bool? I can't find docs about this.

Reply:
Yeah, it seems so. I had to check `if value == 'true'` because `if value` wasn't working properly, so that checks out